src/main/java/org/ays/common/util/validation/NoTrailingOrLeadingSpaces.java

```diff
@@ -0,0 +1,39 @@
+package org.ays.common.util.validation;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to validate a text using {@link NoTrailingOrLeadingSpacesValidator}.
+ */
+@Target(value = ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = NoTrailingOrLeadingSpacesValidator.class)
+public @interface NoTrailingOrLeadingSpaces {
```

Comment:
`NoTrailingOrLeadingSpaces` yerine daha kısa ve net bir isimlendirme yapabilir miyiz?

Örneğin;
```suggestion
@Constraint(validatedBy = NoExtraSpacesValidator.class)
public @interface NoExtraSpaces {
```

Başka öneriler varsa onları değerlendirelim isterim

Reply:
@agitrubard NoOuterSpaces olabilir belki, ne dersin?

Reply:
@egehanasal Jakarta validasyonlarına yakın olabilmesi adına `NotStartOrEndWithSpace` gibi düşünüyorum ama net bir şey bulamadım.

Reply:
NoSpacesAround fikri ile bir commit attim